quickwit/quickwit-cli/src/lib.rs

```diff
@@ -73,13 +76,127 @@ fn config_cli_arg() -> Arg {
         .display_order(1)
 }
 
-fn cluster_endpoint_arg() -> Arg {
-    arg!(--"endpoint" <QW_CLUSTER_ENDPOINT> "Quickwit cluster endpoint.")
-        .default_value("http://127.0.0.1:7280")
-        .env("QW_CLUSTER_ENDPOINT")
-        .required(false)
-        .display_order(1)
-        .global(true)
+fn client_args() -> Vec<Arg> {
+    vec![
+        arg!(--"endpoint" <QW_CLUSTER_ENDPOINT> "Quickwit cluster endpoint.")
+            .default_value("http://127.0.0.1:7280")
+            .env("QW_CLUSTER_ENDPOINT")
+            .required(false)
+            .display_order(1)
+            .global(true),
+        Arg::new("timeout")
+            .long("timeout")
+            .help("Duration of the timeout.")
+            .required(false)
+            .global(true)
+            .display_order(1),
+        Arg::new("connect-timeout")
+            .long("connect-timeout")
+            .help("Duration of the connect timeout.")
+            .required(false)
+            .global(true)
+            .display_order(1),
+    ]
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct ClientArgs {
+    pub cluster_endpoint: Url,
+    /// The outer option represents the presense of the argument,
+    /// the inner option represents the presence of timeout.
+    pub connect_timeout: Option<Option<Duration>>,
```

Comment:
It's weird to have option of option. Is it possible to simplify that?

Reply:
I tried and this seems to be the lesser evil. The issue is that we need to represent 2 situations here. 1) whether or not the user specified the commit parameters and 2) where we have a timeout or we have no timeout. We could have teach CLI the default timeouts but then we wouldn't be able to reuse the same flag `--timeout` for different timeouts such as search timeout in case of search and ingest_timeout in case of ingest, so we would be pushing this complexity to the user.

Reply:
I see. 
But I don't understand why we need an option of option for `connect_timeout`. Why not having `pub const DEFAULT_CLIENT_CONNECT_TIMEOUT: Duration = Duration::from_secs(5);`

The user can of course override this. But I don't see the need of an option. In the client args, we can only have an `Option<Duration>` for this variable. Or am I missing something?

Reply:
The current setting for connect_timeout is not set. So, since I am setting it pretty aggressively down in this PR I wanted to have an ability to undo that if it doesn't work for some reason, which would be expressed as Some(None) on the CLI level. The main reason I am setting it is because I also increase commit timeout all the way up. It would be a bad user experience to wait half an hour for ingest if we cannot even connect properly.

Reply:
I think it was bad to have not set the `connect_timeout` before, you are fixing that so I suggest to remove the option. 5 seconds is maybe a bit too agressive, something like 10 - 15 seconds sounds reasonable.